TCMSApp/src/client/app/defects/defects.controller.js

```diff
@@ -5,24 +5,66 @@
         .module('app.defects')
         .controller('DefectsController', DefectsController);
 
-    DefectsController.$inject = ['logger', '$uibModal', 'getDefects', '$state'];
+    DefectsController.$inject = ['logger', '$uibModal', 'getDefects', '$state', '$resource', 'moment', 'apiUrl'];
     /* @ngInject */
-    function DefectsController(logger, $uibModal, getDefects, $state) {
+    function DefectsController(logger, $uibModal, getDefects, $state, $resource, moment, apiUrl) {
         var vm = this;
-        vm.arrayDefects = getDefects.fakeDefect(10);
-        vm.toogledAll = false;
-        if (vm.arrayDefects) {
-            vm.currentDefect = vm.arrayDefects[0];
-            vm.setCurrentDefect = function (x) {
-                vm.currentDefect = x;
-            };
-        }
-        else {
-            vm.currentDefect = null;
+        var defectsInfo = $resource(apiUrl.defects, {}, {});
+        var id = '';
+        var name = '';
+        var description = '';
+        var priority = '';
+        var whoFind = '';
+        var assignedTo = '';
+        var stepsToReproduce = '';
+        var dateOfDefectCreation = '';
+        var sample = {};
+        vm.arrayDefects = [];
+        vm.statusMessage = 'Loading defects...';
+        vm.countOfDefects = -1;
+        updateDefects();
+        function updateDefects() {
+            defectsInfo.query(function(resp) {
+                vm.countOfDefects = resp.length;
+                for (var i = 0; i < resp.length; i++) {
```

Comment:
why can't be lines 28-51 substituted into `vm.arrayDefects = resp`? it can be done anyway.